Upstream-Status: Submitted [https://github.com/projectacrn/acrn-kernel/compare/acrn_skl_test]
Signed-off-by: Chin Huat Ang <chin.huat.ang@intel.com>

From a2bcb2a7f6f0727d5578b604c1ce04a91f66ce72 Mon Sep 17 00:00:00 2001
From: Zhao Yakui <yakui.zhao@intel.com>
Date: Mon, 11 Mar 2019 10:45:09 +0800
Subject: [PATCH 1/8] Add the plane restrictionfor SKL. Otherwise there is no
 display for SOS/UOS. Setup the virtual display monitor for UOS on SKL

Signed-off-by: He Min<min.he@intel.com>
Signed-off-by: Zhao Yakui <yakui.zhao@intel.com>
---
 drivers/gpu/drm/i915/gvt/display.c   | 9 ++-------
 drivers/gpu/drm/i915/intel_display.c | 4 ++--
 2 files changed, 4 insertions(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/display.c b/drivers/gpu/drm/i915/gvt/display.c
index afe9eaf..260ce33 100644
--- a/drivers/gpu/drm/i915/gvt/display.c
+++ b/drivers/gpu/drm/i915/gvt/display.c
@@ -592,10 +592,8 @@ void intel_vgpu_clean_display(struct intel_vgpu *vgpu)
 {
 	struct drm_i915_private *dev_priv = vgpu->gvt->dev_priv;
 
-	if (IS_BROXTON(dev_priv) || IS_KABYLAKE(dev_priv))
+	if (IS_BROXTON(dev_priv) || IS_KABYLAKE(dev_priv) || IS_SKYLAKE(dev_priv))
 		clean_virtual_monitors(vgpu);
-	else if (IS_SKYLAKE(dev_priv))
-		clean_virtual_dp_monitor(vgpu, PORT_D);
 	else
 		clean_virtual_dp_monitor(vgpu, PORT_B);
 }
@@ -616,11 +614,8 @@ int intel_vgpu_init_display(struct intel_vgpu *vgpu, u64 resolution)
 
 	intel_vgpu_init_i2c_edid(vgpu);
 
-	if (IS_BROXTON(dev_priv) || IS_KABYLAKE(dev_priv))
+	if (IS_BROXTON(dev_priv) || IS_KABYLAKE(dev_priv) || IS_SKYLAKE(dev_priv))
 		return setup_virtual_monitors(vgpu);
-	else if (IS_SKYLAKE(dev_priv))
-		return setup_virtual_monitor(vgpu, PORT_D, GVT_DP_D,
-						resolution, NULL, true);
 	else
 		return setup_virtual_monitor(vgpu, PORT_B, GVT_DP_B,
 						resolution, NULL, true);
diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 223cda4..e449b8d 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -15440,7 +15440,7 @@ static int intel_sanitize_plane_restriction(struct drm_i915_private *dev_priv)
 	unsigned int mask;
 
 	/*plane restriction feature is only for APL and KBL for now*/
-	if (!(IS_BROXTON(dev_priv) || IS_KABYLAKE(dev_priv))) {
+	if (!(IS_BROXTON(dev_priv) || IS_KABYLAKE(dev_priv) || IS_SKYLAKE(dev_priv))) {
 		i915_modparams.avail_planes_per_pipe = 0;
 		DRM_INFO("Turning off Plane Restrictions feature\n");
 	}
@@ -15448,7 +15448,7 @@ static int intel_sanitize_plane_restriction(struct drm_i915_private *dev_priv)
 	mask = i915_modparams.avail_planes_per_pipe;
 
 	/* make sure SOS has a (dummy) plane per pipe. */
-	if ((IS_BROXTON(dev_priv) || IS_KABYLAKE(dev_priv)) &&
+	if ((IS_BROXTON(dev_priv) || IS_KABYLAKE(dev_priv) || IS_SKYLAKE(dev_priv)) &&
 			intel_gvt_active(dev_priv)) {
 		enum pipe pipe;
 
-- 
2.7.4

